### ISSUE TITLE
[CLOSED] Handling of "assert" keyword

### ISSUE THREAD

**Issue by aionbot** [aionnetwork/AVM#72]
**Issue created by [jeff-aion](https://github.com/jeff-aion)** (on Monday Jun 04, 2018 at 17:20 GMT)

The `assert` keyword generates into some rather specialized code which we need to determine how to handle.
When a class uses this keyword, the compiler generates a static field in the output:  `$assertionsDisabled:Z` as well as the routine in the class init routine to populate it from the response of `java/lang/Class.desiredAssertionStatus:()Z` on the instance of the class object, iself.
This, ultimately, is derived from whether or not the JVM was started with the `-ea` argument.
We probably want to force this to be always true either through bytecode manipulation or through our implementation of `avm_desiredAssertionStatus`, in the shadow definition of `Class`.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Tuesday Aug 28, 2018 at 19:38 GMT)

The core of the required work for this item was already done under a8370e6eb7e82305be0424503de20274c4e60dd7, in order to get blake2b working:  the implementation of `Class#avm_desiredAssertionStatus()`.

My work for this item will just involve creating a test to demonstrate how this works and adding some documentation to the `avm_desiredAssertionStatus()` implementation.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Tuesday Aug 28, 2018 at 20:41 GMT)

This depends on #207, since the generated code will call these specialized constructors.
